src/transformation.h

```diff
@@ -0,0 +1,53 @@
+#ifndef TRANSFORMATION_H
+#define TRANSFORMATION_H
+#include <iostream> 
+#include "data_point.h"
+#include "hashing_extractor.h"
+#include <vector>
+namespace usermodel{
+
+enum class TransformationType{
+    TO_LOWER = 1,
+    HASHED_NGRAMS=2,
+    NORMALIZE=3
+};
+
+class Transformation{
+    public:
+        Transformation();
+        Transformation(TransformationType t);
+        Transformation(std::string parameters);
+        ~Transformation();
+        DataPoint get(DataPoint inp);
+        TransformationType getType();
+    protected:
+        HashVectorizer hash_vectorizer;
+        TransformationType type;
```

Comment:
Not initialized in release builds, so will cause issues. Discussed adding a new type (first entry) and setting to NONE. Please check all other simple types, as they need initializing